### DISCUSSION TITLE
Configs: Last round of "final" changes to configs for v6

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#7130] [Community Feedback]
Following #6014, I described the changes in #7110 as the _"final v6 changes to configs"_. That was inaccurate - it was the final changes to configs _before trying them out in real-world projects_ (#6760). Now that we've tried things out, there are a few more changes I think we should make...

Increasing strictness:

* `no-explicit-any`: It was moved from _`recommended`_ to _`strict`_, but I think we should move it back. Most projects shouldn't be using `any`.
* `strict-boolean-expressions`: Switch `allowNullableEnum` to `false` by default, per https://github.com/typescript-eslint/typescript-eslint/discussions/6014#discussioncomment-5774614
    * _(this was originally listed under the following section erroneously - thanks @tylerlaprade for pointing that out!)_

Decreasing strictness:

* `no-confusing-void-expressions`: It was added to _`stylistic-type-checked`_ but I think it's too opinionated for many users, especially for those working in JSX-based frameworks. I think we should remove it from _`stylistic-type-checked`_ for now.
* ~`no-mixed-enums`: It was added to _`recommended`_, but I think it's a rare enough edge case that a large percentage of consumers wouldn't care for the rule. I think we should instead move it to _`strict`_ for now.~ _(this is in `strict-type-checked`, I was wrong)_
* `sort-type-constituents`: It was added to _`stylistic`_. It's pretty nitpicky without actual bug-catching and I think many users would be by it. I think we should remove it from _`stylistic`_ for now.

...anything else that's come up & we should adjust for?

**Comment by JoshuaKGoldberg** [comment]
Coming over from https://github.com/typescript-eslint/typescript-eslint/pull/7138:

* `prefer-string-starts-ends-with`: Perhaps #7140 should be in there?
* Confirmed on `no-confusing-void-expressions`: we don't like it in `stylistic-type-checked`. Maybe `strict-type-checked`, or nothing at all?

**Comment by tylerlaprade** [comment]
Technically the `strict-boolean-expressions: allowNullableEnum` change should be listed under the "Increasing strictness" section. Thanks for including it, by the way!

**Reply by JoshuaKGoldberg** [reply]
Whoops, thanks! Edited :)
